pandas/core/internals/blocks.py

```diff
@@ -1389,6 +1389,25 @@ def shift(self, periods: int, axis: int = 0, fill_value=None):
 
         return [self.make_block(new_values)]
 
+    def _maybe_reshape_where_args(self, values, other, cond, axis):
```

Comment:
assume you are going to share this / refactor at some point. type / document args at some point. also consider making a module level function (as except for the transpose it doesn't need to be a method)

Reply:
yah, i tentatively think we'll be able to use this for Block.putmask too